safenode/src/node/api.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2023 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use std::time::Duration;
+
+use futures::future::select_all;
+use xor_name::XorName;
+
+use super::error::Result;
+use crate::{
+    network::{Cmd, Query, Request},
+    storage::chunks::Chunk,
+};
+
+use super::Node;
+
+impl Node {
```

Comment:
better have a name like `Vault` (or even just DataHolder) if it only handles the data storage duty?
`Node` means it also needs to handle the `network knowledge` and other duties ?